src/ServerConfig.cc

```diff
@@ -106,11 +107,28 @@ ServerConfig::PluginInfo::PluginInfo(const std::string &_entityName,
                        const std::string &_filename,
                        const std::string &_name,
                        const sdf::ElementPtr &_sdf)
-  : dataPtr(new ServerConfig::PluginInfoPrivate(_entityName, _entityType,
-                                  _filename, _name))
+  : dataPtr(new ServerConfig::PluginInfoPrivate())
 {
   if (_sdf)
+  {
     this->dataPtr->sdf = _sdf->Clone();
+    this->dataPtr->plugin.Load(this->dataPtr->sdf);
+  }
+  this->dataPtr->plugin.SetName(_name);
+  this->dataPtr->plugin.SetFilename(_filename);
+  this->dataPtr->filename = _filename;
+  this->dataPtr->name = _name;
+  this->dataPtr->entityName = _entityName;
+  this->dataPtr->entityType = _entityType;
+}
+
+//////////////////////////////////////////////////
+ServerConfig::PluginInfo::PluginInfo(const std::string &_entityName,
+                       const std::string &_entityType,
+                       const sdf::Plugin &_plugin)
+  : dataPtr(new ServerConfig::PluginInfoPrivate(_entityName, _entityType,
```

Comment:
Also here `make_unique` can be used

Reply:
Done in [afd30fb](https://github.com/gazebosim/gz-sim/pull/1352/commits/afd30fbb79c7c81cf9d7bb056179fa6cd92a0d94)